### ISSUE TITLE
Footnotes

### ISSUE THREAD

**Issue by millerjs2** [xenocrat/chyrp-lite#187]
Hello,
Is it possible to use footnotes in posts or pages? I've tried using a few different methods I've found online, but none of them seem to work. Is this supported, and if not, I would like to request that this capability be added. This is how they work in the extended markdown spec:

https://www.markdownguide.org/extended-syntax/#footnotes

**Comment by xenocrat** [comment]
Hello there,

Chyrp Lite is using [cebe/markdown](https://github.com/cebe/markdown) to provide "[Github flavored Markdown](https://github.github.com/gfm/)" which does not support footnotes. Adding the capability is quite a task, but I will add it to my list of features to investigate.

In the meantime, if Markdown footnotes are a must-have, you could consider disabling the built-in Markdown support and adding your own by [writing a module](https://github.com/xenocrat/chyrp-lite/wiki/Making-Your-First-Module). You could use [michelf/php-markdown](https://github.com/michelf/php-markdown) which does support footnotes.

**Comment by xenocrat** [comment]
A brief update: I am evaluating [CommonMark](https://github.com/thephpleague/commonmark) as an alternative Markdown implementation. This would allow Chyrp Lite to support footnotes. There are performance implications to switching Markdown implementation, so I will need to investigate this thoroughly before I decide. I’ll update this issue once I have time to look into it.

**Comment by millerjs2** [comment]
Thank you, I appreciate it! And thanks again for the good work on this project, it's easily my favorite blogging platform.

**Comment by xenocrat** [comment]
An update: I decided not to use thephpleague/commonmark implementation, partly due to performance concerns but mostly because it is completely impractical to use outside the Composer environment. However I have patched in footnote support to the cebe/markdown implementation currently used by Chyrp Lite. As long as I don't run into any terrible bugs during testing, footnote support will be available in the next release.
